registry/storage/layerwriter.go

```diff
@@ -14,9 +14,11 @@ import (
 	"github.com/docker/docker/pkg/tarsum"
 )
 
-// layerUploadController is used to control the various aspects of resumable
+var _ distribution.LayerUpload = &layerWriter{}
```

Comment:
Having layerReader and layerUpload was inconsistent naming vs fileReader and fileWriter. Also, Go uses "Reader" and "Writer" consistently. Given the interfaces implemented, layerWriter seems like a much more appropriate and consistent name.